src/main/cpp/kernels/SystemML.cu

```diff
@@ -1961,3 +1961,100 @@ extern "C" __global__ void matrix_sigmoid_f(float *A, float *C,
                                          unsigned int size) {
   matrix_sigmoid(A, C, size);
 }
+
+// We can later fold it in our reduce method
+template <typename T>
+__device__ void compute_nnz(
```

Comment:
This should reuse our templatized reduction method, as you mentioned in the comment above.